### ISSUE TITLE
Foldername AUX is reserved on NTFS drives

### ISSUE THREAD

**Issue by cserteGT3** [tlienart/Xranklin.jl#256]
I thought that I will go for a test run, but it fails to `add`, `dev` or even clone the package. Below is the output of git bash cloning. Julia Pkg fails with the same error.

```
$ git clone https://github.com/tlienart/Xranklin.jl
Cloning into 'Xranklin.jl'...
remote: Enumerating objects: 15941, done.
remote: Counting objects: 100% (3161/3161), done.
remote: Compressing objects: 100% (665/665), done.
remote: Total 15941 (delta 1853), reused 3086 (delta 1830), pack-reused 12780
Receiving objects: 100% (15941/15941), 28.83 MiB | 2.81 MiB/s, done.
Resolving deltas: 100% (9163/9163), done.
error: invalid path 'test/indir/aux/literate.jl'
fatal: unable to checkout working tree
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry with 'git restore --source=HEAD :/'
```

Do you have any idea what happened there? On github I can read that file and download it as well.

**Comment by tlienart** [comment]
🧐

very strange, could you try forking the repo, and adding your fork? 

Separately could you try adding a previous commit?

There are several active users of Xranklin so this is a bit unexpected. Oh and what OS are you on? Not sure that would make a difference though..

**Comment by cserteGT3** [comment]
With the help of a colleague I found out, that the problem is the `aux` folder, which is a reserved name on NTFS drives.
[Source](https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file):

> Do not use the following reserved names for the name of a file:
> CON, PRN, **AUX**, NUL, COM0, COM1, COM2, COM3, COM4, COM5, COM6, COM7, COM8, COM9, LPT0, LPT1, LPT2, LPT3, LPT4, LPT5, LPT6, LPT7, LPT8, and LPT9. Also avoid these names followed immediately by an extension; for example, NUL.txt and NUL.tar.gz are both equivalent to NUL. For more information, see [Namespaces](https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file#win32-file-namespaces).

It is confirmed by:
* My drive is indeed formatted as NTFS. (On Windows 11)
* Colleague is on Linux. Checkout failed for him on an NTFS drive, but worked on an ext4 drive.
* Under WSL, the checkout worked, because it is on a partition formatted as ext4.

I tested on my [fork](https://github.com/cserteGT3/Xranklin.jl), that after renaming the `aux` folder to `aux2`, I can checkout on NTFS drives.

So I suggest renaming the aux folder to something else. Sadly I cannot really open a PR, because I can not checkout to the repo if it contains the aux path :D
